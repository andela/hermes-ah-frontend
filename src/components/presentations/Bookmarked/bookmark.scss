.bookmark {
    button {
        font-size: 20px;
        background: #fff;
        border: none;
        color: #0e71eb;
        cursor: pointer;
    }
}