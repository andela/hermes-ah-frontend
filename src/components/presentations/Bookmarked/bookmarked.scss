@import '../../../assets/stylesheets/index.scss';
h3, p {
    font-family: $font-stack;
}

.bookmark-articles {
    width: 70%;
    overflow: auto;
    margin: 0 auto;

    .header-card {
        width: 80%;
        margin: 20px 0;
    }
    .icon {
        color: $primary-color;
        padding: 0 10px;
    }
}
