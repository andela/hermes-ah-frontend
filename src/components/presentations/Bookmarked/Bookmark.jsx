import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './bookmark.scss';

class Bookmark extends Component {
  handleBookmark = event => {
    event.preventDefault();
    const { articleId } = this.props;

    const { bookmark } = this.props;
    bookmark(articleId);
  };

  render() {
    return (
      <div className="bookmark">
        <button type="button" onClick={this.handleBookmark}>
          <i className="far fa-bookmark" />
        </button>
      </div>
    );
  }
}

Bookmark.propTypes = {
  articleId: PropTypes.string.isRequired,
  bookmark: PropTypes.func.isRequired,
};

export default Bookmark;
