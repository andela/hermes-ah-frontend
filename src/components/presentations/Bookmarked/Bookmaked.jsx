import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article/Article';
import './bookmarked.scss';

const Bookmarked = ({ bookmarkedArticles }) => {
  const { articles } = bookmarkedArticles;

  return (
    <div className="bookmark-articles">
      <div className="header-card">
        <div className="icon">
          <i className="fas fa-bookmark" />
        </div>
        <div className="rider">
          <h2>Bookmarks - Articles here are private and only visible to you</h2>
        </div>
      </div>
      {articles.map(item => (
        <Article
          key={item.article_id}
          title={item.Article.title}
          abstract={item.Article.abstract}
          image={item.Article.image_url}
          readTime={item.Article.reading_time}
        />
      ))}
    </div>
  );
};

Bookmarked.propTypes = {
  bookmarkedArticles: PropTypes.shape().isRequired,
};

export default Bookmarked;
