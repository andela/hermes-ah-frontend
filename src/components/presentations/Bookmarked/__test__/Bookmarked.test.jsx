import React from 'react';
import { shallow } from 'enzyme';

import Bookmarked from '../Bookmaked';

const bookmarkedArticles = {
  articles: [
    {
      id: 1,
      title: 'This is a cat',
      abstract: 'March 3 2019',
      image:
        'https://res.cloudinary.com/walsam/image/upload/v1556790944/brain-2062057_640.jpg',
      readTime: 3,
    },
    {
      id: 2,
      title: 'This is a dog',
      abstract: 'March 23 2019',
      image:
        'https://res.cloudinary.com/walsam/image/upload/v1556790944/brain-2062057_640.jpg',
      readTime: 3,
    },
    {
      id: 3,
      title: 'This is a fish',
      abstract: 'March 13 2019',
      image:
        'https://res.cloudinary.com/walsam/image/upload/v1556790944/brain-2062057_640.jpg',
      readTime: 3,
    },
  ],
};

const BookmakedComponent = (
  <div>
    <Bookmarked bookmarkedArticles={bookmarkedArticles} />
  </div>
);

describe('<ArticleCard />', () => {
  it('should render bookmark component', () => {
    const wrapper = shallow(BookmakedComponent);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });
});
