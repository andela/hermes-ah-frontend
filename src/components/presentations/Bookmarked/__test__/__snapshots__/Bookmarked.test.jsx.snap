// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> should render bookmark component 1`] = `ShallowWrapper {}`;
